### PR TITLE
Jerry

### PR DESCRIPTION
**Pull request by sshJerry**
Customer related functionality

### CODE REVIEW

File: src/com/main/service/UserService.java [hunk 1]
```diff
@@ -43,6 +43,7 @@ public void createAccount() {
 			
 			// Option 1 - Customer
 			if (userInput == 1) {
+
```
Comment (Kev-Index) [inline src/com/main/service/UserService.java]: Is there a reason to add a line here?

File: src/com/main/service/CustomerService.java [hunk 2]
```diff
@@ -3,31 +3,78 @@
...
+		System.out.println("DEBUG: Connectioned"); 
+		System.out.println("Enter your username");
+		String customerLoginUsername = sc.nextLine();
+		boolean isValidUsername = CustomerUtility.validateUsername(db.getAllCustomers(),customerLoginUsername);
+		if(!isValidUsername) 
+			System.out.println("Incorrect Credentials");
+
+		System.out.println("Enter your password");
+		String customerLoginPassword = sc.nextLine();
+		boolean isValidPassword = CustomerUtility.validatePassword(db.getAllCustomers(),customerLoginPassword);
```
Comment (Kev-Index) [inline src/com/main/service/CustomerService.java]: The method called here will only check if the password exists in the db, not if the previously entered username matches this password

File: src/com/utility/CustomerUtility.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+			if(c.getUsername().equals(customerLoginUsername)) {
+				isValid = true;
+				break;
+			}
+		}
+		return isValid;*/
+	}
+
+	public static boolean validatePassword(List<Customer> allCustomers, String customerLoginPassword) {
+		return allCustomers.stream().map(c->c.getPassword()).collect(Collectors.toList()).contains(customerLoginPassword);
```
Comment (Jlpowellv) [inline src/com/utility/CustomerUtility.java:23]: this still does not validate that the password and the username belong to the same user

File: src/com/main/service/CustomerService.java [hunk 4]
```diff
@@ -3,31 +3,77 @@
...
+			
+			System.out.println("Enter your password");
+			String customerLoginPassword = sc.nextLine();
+			boolean isValidPassword = CustomerUtility.validatePassword(db.getAllCustomers(),customerLoginPassword);
+			if(!isValidPassword || !isValidUsername) {
+				System.out.println("Incorrect Credentials");
+			}
+			
+			if(isValidUsername && isValidPassword) {
+				System.out.println("Signed in!");
```
Comment (Jlpowellv) [inline src/com/main/service/CustomerService.java:72]: this should send you to the order menu and then break afterwards, so the person can order some food ya know (we also have to store the customer id somehow)

File: src/com/utility/CustomerUtility.java [hunk 5]
```diff
@@ -9,18 +9,14 @@ public class CustomerUtility {
...
 	}
 
-	public static boolean validatePassword(List<Customer> allCustomers, String customerLoginPassword) {
-		return allCustomers.stream().map(c->c.getPassword()).collect(Collectors.toList()).contains(customerLoginPassword);
+	public static boolean validatePassword(List<Customer> allCustomers, String customerLoginPassword, String customerLoginUsername) {
+		return allCustomers.stream()
+				.filter(c->c.getUsername().equals(customerLoginUsername))
+				.map(c->c.getPassword())
+				.collect(Collectors.toList())
+				.contains(customerLoginPassword);
```
Comment (Kev-Index) [inline src/com/utility/CustomerUtility.java:19]: Looks good now! Just gotta resolve merge conflicts

**Review by jayharris14 — APPROVED** [review]

**Review by jayharris14 — APPROVED** [review]

**Review by Kev-Index — APPROVED** [review]
LOOKING GOOD

**Pull request merged**